yt/frontends/ramses/io.py

```diff
@@ -118,6 +117,50 @@ def _ramses_particle_file_handler(fname, foffsets, data_types, subset, fields, c
     return tr
 
 
+def _ramses_particle_csv_file_handler(particle, subset, fields, count):
+    """General file handler for csv file, called by _read_particle_subset
+
+    Parameters
+    ----------
+    particle : ``ParticleFileHandler``
+        the particle class we want to read
+    subset: ``RAMSESDomainSubset``
+        A RAMSES domain subset object
+    fields: list of tuple
+        The fields to read
+    count: integer
+        The number of elements to count
+    """
+    from yt.utilities.on_demand_imports import _pandas as pd
+
+    tr = {}
+    ds = subset.domain.ds
+    current_time = ds.current_time.in_units("code_time").v
+    foffsets = particle.field_offsets
+    fname = particle.fname
+
+    for field in sorted(fields, key=lambda a: foffsets[a]):
+        ind = foffsets[field]
+        dat = pd.read_csv(
+            fname, delimiter=",", usecols=[ind], skiprows=2, header=None
+        )  # read only selected fields
```

Comment:
I'm wondering if we could call `pd.read_csv` just once ahead of the loop instead of doing multiple (costly) reading operations ? If there was a chance of memory errors, I would expect the current implementation to hit those any way.

Reply:
ok I change now. 
But, it's far from being optimized : I nead to read the entire file in `_read_part_csv_file_descriptor` to get the number of lines (to then initialize `self.local_particle_count`). And then I read it again (but only the columns i'm interested in)

Reply:
I think reading is a csv file is to different from readinf the fortan binay file, so thying to use the same structure is not ideal